api/api.go

```diff
@@ -359,15 +360,33 @@ func (api *API) delSilence(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
-	sils, err := api.silences.All()
+	var (
+		num    = r.FormValue("n")
+		offset = r.FormValue("offset")
```

Comment:
I had this discussion with stuart in person. He made a point that for the order of a few thousand silences, this is unlikely to be a problem in the backend but gives huge benefits in the frontend right now.

Another iteration approach requires some fundamental changes in the backend storage, which isn't in its final form right now and doesn't support it.